packages/amplify-category-function/src/index.ts

```diff
@@ -156,6 +161,11 @@ export async function initEnv(context) {
         _.set(teamProviderInfo, [envName, 'categories', categoryName, resourceName], tpiResourceParams);
         _.set(amplifyMeta, [categoryName, resourceName, 's3Bucket'], s3Bucket);
       }
+
+      // if the function has secrets, set the appId key in team-provider-info
+      if (!!getLocalFunctionSecretNames(resourceName, { fromCurrentCloudBackend: true }).length) {
+        _.set(teamProviderInfo, [envName, 'categories', categoryName, resourceName, secretsPathAmplifyAppIdKey], getAppId());
```

Comment:
nit: Would be good err on the side of caution when accessing side effects.  would be good to validate the return value of getAppId() and throw an exception if not valid.

Reply:
getAppId already throws an error if a value can't be determined